tests/mock_smtp.py

```diff
@@ -70,7 +70,7 @@ def loop(self):
         exit.
         """
         while len(asyncore.socket_map):
-            asyncore.loop(timeout=1, use_poll=True)
+            asyncore.loop(timeout=0, use_poll=True)
```

Comment:
Can you try changing the timeout something like 0.25 or 0.1 and see if that prevents you from getting the "bad file descriptor" error?  I think any timeout is better than a 0 timeout.  Ideally, it would be as large as possible and still allow a graceful exit.